mlir/lib/Transforms/LocationSnapshot.cpp

```diff
@@ -140,7 +140,7 @@ struct LocationSnapshotPass
 
   void runOnOperation() override {
     Operation *op = getOperation();
-    if (failed(generateLocationsFromIR(fileName, op, OpPrintingFlags(), tag)))
+    if (failed(generateLocationsFromIR(fileName, op, flags, tag)))
```

Comment:
This actually makes the pass not serializable, which I don't think is acceptable. Can we split the OpPrintingFlag option into "proper" options (strings and boolean) and construct the OpPrintingFlag on the fly?